packages/nextjs/scripts/generate-types.ts

```diff
@@ -0,0 +1,281 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+/**
+ * Generate `@sentry/nextjs` types.
+ *
+ * This file (compiles to) a script which generates a merged set of exports for the nextjs SDK, for the ultimate purpose
+ * of generating the SDK's types. The `module` and `browser` entries in `package.json` allow different files to serve as
+ * the SDK's entry point depending on environment, but there is no such bifurcation when it comes to types. Leaving the
+ * `types` entry out causes TypeScript to search for a declaration file whose name matches each of `module` and
+ * `browser` values, respectively, and that allows compilation to happen (yay!) but without either `types` or `main`
+ * entries in `package.json`, VSCode is unable to resolve the SDK for code completion, Intellisense, etc. (boo!). The
+ * `types` entry can't be an array, but even if it could, that wouldn't handle the case of namespace clashes
+ * (`@sentry/node.Integrations` is not the same as `@sentry/react.Integrations`, for instance - which one wins?). So
+ * there needs to be a single source of truth, generated by semi-intelligently merging the exports from the two packages
+ * (by way of the two different, environment-specific index files), such that types can then be generated from that
+ * single file.
+ *
+ * Known limitations:
+ *
+ *  - In a small handful of mostly-non-user-relevant spots, there's no easy way to resolve the conflict (for example,
+ *    which `flush` and `close` methods should be exported, node's or react's?) and so those types have been omitted
+ *    from this file. The correct methods are stil exported in their respective environments - the JS works, in other
+ *    words - but they won't appear in the types. The one exception to this is the `init` method, because it's too
+ *    important to leave out. For the moment, it's using the node version as the type in both environments. (TODO: Make
+ *    a generic version of `init`, and possibly the other clashes, just for use in types.)
+ *
+ * The file that this script's compiled version generates is `/src/types.ts`.
+ */
+
+// TODO - should we be importing from the two index files instead?
+
+import * as nodeSDK from '@sentry/node';
```

Comment:
This is a really nice piece of code generation, but I'm kinda afraid to release it to the public without at least some e2e tests (eg. the one as we have for browser with webpack compilation).